README.md

```diff
@@ -9,270 +9,314 @@ Benefits
 
-```toml
-[tool.uv.workspace]
-members = ["packages/*"]
+   ```toml
+   [tool.uv.sources]
+   ...
+   nomad-parser-plugins-electronic = { workspace = true }
+   ```
 
-[tool.uv.sources]
-nomad-lab = { workspace = true }
-nomad-parser-plugins-electronic = { workspace = true }
-nomad-parser-plugins-atomistic = { workspace = true }
-nomad-parser-plugins-workflow = { workspace = true }
-nomad-parser-plugins-database = { workspace = true }
-...
-```
+   If you're developing a plugin **not** listed under `[project.dependencies]`, you 
+   must first add it as a dependency. After adding the dependencies, update the 
+   `[tool.uv.sources]` section in your `pyproject.toml` file to reflect the new 
+   plugins. You can either add it manually abd run `uv sync`:
 
-> [!NOTE]
-> If you're developing a plugin not listed under [project.dependencies], you must first add it as a dependency. You can do this with uv:
+   ```toml
+   [project]
+   dependencies = [
+   ...
+   "nomad-measurements",
+   ]
+   
+   [tool.uv.sources]
+   ...
+   nomad-measurements = { workspace = true }
+   ```
 
-```bash
-uv add nomad-measurements
-uv add https://github.com/FAIRmat-NFDI/nomad-parser-vasp --branch develop
-```
+   Or, you can use `uv add` which adds the dependency and the source in `pyproject.toml` 
+   and sets up the environment:
 
-After adding the dependencies, update the [tool.uv.sources] section in your pyproject.toml file to reflect the new plugins:
+   ```bash
+   uv add nomad-measurements
+   ```
 
-```toml
-dependencies = [
-  ...
-  "nomad-measurements",
-  "nomad-parser-vasp",
-]
+   > [!NOTE]
+   > You can also use `uv` to install a specific branch of the plugin submodule.   
+   > ```bash
+   > uv add https://github.com/FAIRmat-NFDI/nomad-measurements --branch <specific-branch-name>
+   > ```
 
-[tool.uv.sources]
-...
-nomad-measurements = { workspace = true }
-nomad-parser-vasp = { workspace = true }
-```
+4. Install dependencies
 
-3. Install dependencies
+   Run the following command to install all dependencies, including the local packages in editable mode:
 
-Run the following command to install all dependencies, including the local packages in editable mode:
+   ```bash
+   uv sync
+   ```
 
-```bash
-uv sync
-```
+   > [!NOTE]
+   > `uv sync` and `uv run` automatically manages the virtual environment for you. 
+   > There's no need to manually create or activate a venv.
+   > Any `uv run` commands will automatically use the correct environment by default.
+   > Read more about `uv` commands to manage the dependencies [here](https://docs.astral.sh/uv/concepts/projects/#managing-dependencies). 
 
-> [!NOTE]
-> `uv sync` and `uv run` automatically manages the virtual environment for you. There's no need to manually create or activate a venv.
-> Any `uv run` commands will automatically use the correct environment by default.
+5. Running `nomad` api app.
 
-4. Running `nomad` api app.
```

Comment:
Based on Alvin's feedback, I think we can remove `nomad admin run appworker` and `yarn start` from the setup part and move it to day to day dev section. We could instead say, the setup is done here, we can move on to day to day development section now?

Reply:
Sounds good!